_user/rhealstone/deadlock-break/main.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Rhealstone benchmark
+ *
+ * Task switch
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdatomic.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+#include <board_config.h>
+#include <rhcommon.h>
+
+
+#define BENCHMARKS 1000
+
+#define ONE_TICK     26000 /* Number dependent on CPU. For loop to this count must be longer than sleep period (1 ms) */
+#define ONE_TICK_AVG 24990 /* For loop up to this count takes 1ms */
+
+bool deadBrk;
+atomic_int count = 0;
+atomic_bool done = false;
```

Comment:
I think this should be `volatile`, but I'm not sure what exact effect it has on `atomic_`s